rocAL/include/readers/image/caffe2_lmdb_record_reader.h

```diff
@@ -116,4 +113,22 @@ class Caffe2LMDBRecordReader : public Reader {
     MDB_txn* _read_mdb_txn;
     MDB_cursor* _read_mdb_cursor;
     void open_env_for_read_image();
+    unsigned _shard_start_idx;
```

Comment:
I feel that all these variables that are common to the derived classes can be made as protected members of the base class which can help us reduce the .h file changes, Please check

Reply:
Check with Shobana once